lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -713,7 +716,11 @@ Future<DomXMLHttpRequest> domHttpRequest(String url, {String? responseType}) {
   }));
 
   xhr.addEventListener('error', allowInterop(completer.completeError));
-  xhr.send();
+  if (sendData != null) {
```

Comment:
Do you really need this null check here? There is one inside the method itself. `xhr.send()` is equivalent to `xhr.send(null)` anyway, right?